### PR TITLE
[core] Fixed C++11 build (win)

### PR DESCRIPTION
**Pull request by maxsharabayko**
Fixed some places where pthreads was still in use.

### TODO

- [x] Handle the case `if (pthread_equal(pthread_self(), m_WorkerThread))`

### CODE REVIEW

File: srtcore/queue.cpp [hunk 1]
```diff
@@ -1529,15 +1529,15 @@ void CRcvQueue::stopWorker()
     m_bClosing = true;
 
     // Sanity check of the function's affinity.
-    if (pthread_equal(pthread_self(), m_WorkerThread))
-    {
-        LOGC(mglog.Error, log << "IPE: RcvQ:WORKER TRIES TO CLOSE ITSELF!");
-        return; // do nothing else, this would cause a hangup or crash.
-    }
+    //if (pthread_equal(pthread_self(), m_WorkerThread))
```
Comment (ethouris) [inline srtcore/queue.cpp:1532]: What's wrong with replacing it with
```
if (this_thread::get_id() == m_WorkerThread.get_id())
```
You'll just have to implement `this_thread` thing for POSIX.

File: srtcore/sync.h [hunk 2]
```diff
@@ -75,13 +77,14 @@ class Duration
     inline bool operator<(const Duration& rhs) const { return m_duration < rhs.m_duration; }
 
 public: // Assignment operators
-    inline void operator*=(const double mult) { m_duration = static_cast<int64_t>(m_duration * mult); }
+    inline void operator*=(const int64_t mult) { m_duration = static_cast<int64_t>(m_duration * mult); }
```
Comment (ethouris) [inline srtcore/sync.h:80]: Why is this required?

Comment (maxsharabayko) [reply srtcore/sync.h:80]: This was changed for compatibility with C++11 imlementation, that does not define multiplications by float point types.

File: srtcore/queue.cpp (hunk 1 above)
Comment (maxsharabayko) [reply srtcore/queue.cpp]: Created a dedicated function.
```c++
bool CheckIfThisThread(const CThread& th);
```

**Comment by maxsharabayko** [comment]
A test case just to confirm `srt::sync::CThread::id::operator==(..)` is called properly.
```c++
void* func(void* th)
{
    CThread* thread = reinterpret_cast<CThread*>(th);
    cout << "Thread check. Expected true. Result: " << (thread->get_id() == srt::sync::this_thread::get_id()) << endl;
    return NULL;
}

TEST(SyncThread, OperatorCompare)
{
    CThread th;
    StartThread(th, &func, (void*) &th, "Name");
    cout << "Main check. Expected false. Result: " << (th.get_id() == srt::sync::this_thread::get_id()) << endl;

    th.join();
}
```
